packages/react-mongo/react-mongo.ts

```diff
@@ -0,0 +1,118 @@
+import { Meteor } from 'meteor/meteor'
+import { Mongo } from 'meteor/mongo'
+import { Tracker } from 'meteor/tracker'
+import { EJSON } from 'meteor/ejson'
+import { useEffect, useMemo, useReducer, useRef, DependencyList } from 'react'
+
+const fur = (x: number): number => x + 1
+const useForceUpdate = () => useReducer(fur, 0)[1]
+
+type useSubscriptionRefs = {
+  facade: Meteor.SubscriptionHandle,
+  subscription?: Meteor.SubscriptionHandle,
+  computation?: Tracker.Computation,
+  updateOnReady: boolean,
+  isReady: boolean,
+  params: {
+    name?: string,
+    args: any[]
+  }
+}
+
+const useSubscriptionClient = (name?: string, ...args: any[]) => {
+  const forceUpdate = useForceUpdate()
+
+  const ref = useRef<useSubscriptionRefs>({
+    facade: {
+      stop () {
+        refs.subscription?.stop()
+      },
+      ready () {
+        // Ready runs synchronously with render, and should not
+        // create side effects.
+        refs.updateOnReady = true
+
+        return (refs.subscription && EJSON.equals(refs.params, { name, args }))
+          ? refs.isReady
+          : false
+      }
+    },
+    updateOnReady: false,
+    isReady: false,
+    params: {
+      name,
+      args
+    }
+  })
+  const refs: useSubscriptionRefs = ref.current
+
+  if (!EJSON.equals(refs.params, { name, args })) {
+    refs.updateOnReady = false
+    refs.isReady = false
+  }
+
+  useEffect(() => {
+    refs.computation = Tracker.nonreactive(() => (
```

Comment:
Ah interesting, it's this computation you need, and not the inner one (from `Tracker.autorun`)? Then I also need to fix that in my package.

Reply:
It seems to be the same; [it just passes on the return value](https://github.com/meteor/meteor/blob/master/packages/tracker/tracker.js#L603).

Reply:
yeah, I used to set it inside the function - 6 of one, half dozen of the other.